src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+  a?: number;
+  b?: number;
+  c?: number;
+  radius?: number;
+  width?: number;
+  height?: number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
 
-export class Circle {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(`sides of a ${this.shape} should have positive numbers`);
+    }
 
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(
+        `sides ${a}, ${b}, and ${c} cannot form a triangle.`,
+      );
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
 
+    if (radius <= 0) {
+      throw new Error(`radius of a ${this.shape} should have positive numbers`);
+    }
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * (this.radius ** 2)) * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape: Shape;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    this.shape = 'rectangle';
+
+    if (width <= 0 || height <= 0) {
```

Comment:
Instead of using the 'width' and 'height' variables directly which are undefined in this scope, you should use 'this.width' and 'this.height' that refer to the rectangle's width and height values.